tests/unit/project_loader/extensions/test_utils.py

```diff
@@ -267,6 +274,41 @@ class ExtensionMergeTest(ExtensionTestBase):
                     "command": "echo 'hello'",
                     "environment": {"FOO": "BAR", "TEST_EXTENSION": 1},
                 },
+                "part_definition": textwrap.dedent(
+                    """\
+                    plugin: nil
+                    """
+                ),
+                "expected_part_definition": {"plugin": "nil", "prime": [], "stage": []},
+            },
+        ),
+        (
+            "merge build environment",
+            {
+                "app_definition": textwrap.dedent(
+                    """\
+                    command: echo 'hello'
+                    extensions: [buildenvironment]
+                    """
+                ),
+                "expected_app_definition": {"command": "echo 'hello'"},
+                "part_definition": textwrap.dedent(
+                    """\
+                    plugin: nil
+                    build-environment:
+                      - PATH: "$PATH:/foo"
```

Comment:
s/foo/part-bin (or part-path), keep the suffix consistent with the part and extension suffix

Reply:
good idea, will do